byzcoin/service.go

```diff
@@ -608,6 +608,14 @@ func (s *Service) GetSignerCounters(req *GetSignerCounters) (*GetSignerCountersR
 // GetUpdates returns instances that have a newer versions than the ones
 // passed to it.
 func (s *Service) GetUpdates(pr *GetUpdatesRequest) (*GetUpdatesReply, error) {
+	s.catchingLock.Lock()
+	s.updateTrieLock.Lock()
+
+	defer func() {
+		s.updateTrieLock.Unlock()
+		s.catchingLock.Unlock()
+	}()
```

Comment:
How about:
```
s.catchingLock.Lock()
defer s.catchingLock.Unlock()

s.updateTrieLock.Lock()
defer s.updateTrieLock.Unlock()
```
This leaves it to Go to handle calling the `Unlock()` in the reverse order :)